### PR TITLE
Add pallet names to Events

### PR DESCRIPTION
**Pull request by ECJ222**
This PR changes the pallet events variants from tuples to structs with named fields

Part implementation for #9903, #9993, #10184, #10206, #10257 

## Pallets with changes
- scheduler
- society
- system
- treasury
- transaction-storage
- utility

I also made a few changes to the following pallets after I changed the pallet events variants from tuples to structs with named fields in the `system` pallet. i.e NewAccount, KilledAccount, Remarked.

- balances
- contracts
- offences
- transaction-payment

polkadot companion: https://github.com/paritytech/polkadot/pull/4339

Polkadot address: 13g44iYtkxLZ2tkp6iTJiqs7yEiEr4bVsPZTueWDjd6Bor1P

### CODE REVIEW

File: frame/society/src/lib.rs [hunk 1]
```diff
@@ -476,41 +476,41 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config<I>, I: 'static = ()> {
 		/// The society is founded by the given identity. \[founder\]
```
Comment (KiChjang) [inline frame/society/src/lib.rs:478]:
```suggestion
		/// The society is founded by the given identity.
```

Same for all other event variants.

Comment (ECJ222) [reply frame/society/src/lib.rs:478]: Okay @KiChjang, will do 👍

**Comment by athei** [comment]
CI isn't happy. You need to fix up some tests. I think those are the ones which aren't run when you run tests from the repo root.

**Comment by athei** [comment]
It seems that some of the events are also used by polkadot. You will need to create a [companion](https://github.com/paritytech/substrate/blob/master/docs/CONTRIBUTING.adoc#updating-polkadot-as-well) in order to pass CI.

File: frame/system/src/lib.rs [hunk 2]
```diff
@@ -486,26 +486,26 @@ pub mod pallet {
...
 		/// `:code` was updated.
 		CodeUpdated,
-		/// A new \[account\] was created.
-		NewAccount(T::AccountId),
-		/// An \[account\] was reaped.
-		KilledAccount(T::AccountId),
-		/// On on-chain remark happened. \[origin, remark_hash\]
-		Remarked(T::AccountId, T::Hash),
+		/// A new account was created.
+		NewAccount { new_account: T::AccountId },
```
Comment (bkchr) [inline frame/system/src/lib.rs:504]:
```suggestion
		NewAccount { account: T::AccountId },
```

File: frame/system/src/lib.rs [hunk 3]
```diff
@@ -486,26 +486,26 @@ pub mod pallet {
...
+		/// An account was reaped.
+		KilledAccount { reaped_account: T::AccountId },
+		/// On on-chain remark happened.
+		Remarked { origin: T::AccountId, remarked_hash: T::Hash },
```
Comment (bkchr) [inline frame/system/src/lib.rs:508]:
```suggestion
		KilledAccount { account: T::AccountId },
		/// On on-chain remark happened.
		Remarked { sender: T::AccountId, hash: T::Hash },
```

File: frame/system/src/lib.rs [hunk 4]
```diff
@@ -486,26 +486,26 @@ pub mod pallet {
...
 
 	/// Event for the System pallet.
 	#[pallet::event]
 	pub enum Event<T: Config> {
-		/// An extrinsic completed successfully. \[info\]
-		ExtrinsicSuccess(DispatchInfo),
-		/// An extrinsic failed. \[error, info\]
-		ExtrinsicFailed(DispatchError, DispatchInfo),
+		/// An extrinsic completed successfully.
+		ExtrinsicSuccess { extrinsic_info: DispatchInfo },
```
Comment (bkchr) [inline frame/system/src/lib.rs:498]:
```suggestion
		ExtrinsicSuccess { dispatch_info: DispatchInfo },
```

File: frame/system/src/lib.rs [hunk 5]
```diff
@@ -486,26 +486,26 @@ pub mod pallet {
...
+		/// An extrinsic failed.
+		ExtrinsicFailed { extrinsic_error: DispatchError, extrinsic_info: DispatchInfo },
```
Comment (bkchr) [inline frame/system/src/lib.rs:500]:
```suggestion
		ExtrinsicFailed { dispatch_error: DispatchError, dispatch_info: DispatchInfo },
```

**Comment by ECJ222** [comment]
I have made the changes @athei @bkchr.

File: frame/transaction-storage/src/lib.rs [hunk 6]
```diff
@@ -313,9 +313,9 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
 		/// Stored data under specified index.
-		Stored(u32),
+		Stored { stored_index: u32 },
```
Comment (athei) [inline frame/transaction-storage/src/lib.rs:316]:
```suggestion
		Stored { index: u32 },
```

File: frame/transaction-storage/src/lib.rs [hunk 7]
```diff
@@ -313,9 +313,9 @@ pub mod pallet {
...
 		/// Renewed data under specified index.
-		Renewed(u32),
+		Renewed { renewed_index: u32 },
```
Comment (athei) [inline frame/transaction-storage/src/lib.rs:318]:
```suggestion
		Renewed { index: u32 },
```

**Review by athei — COMMENTED** [review]
Please run `cargo +nightly fmt` to satisfy the CI. Make sure you have a recent nightly installed.

**Review by bkchr — APPROVED** [review]

**Comment by ECJ222** [comment]
> cargo +nightly fmt

I have made those changes, Thanks @athei.

**Comment by KiChjang** [comment]
@ECJ222 CI is still failing on `cargo fmt`, did you use the latest nightly rust to run it?

**Comment by ECJ222** [comment]
> @ECJ222 CI is still failing on `cargo fmt`, did you use the latest nightly rust to run it?

@KiChjang yes I did, I used `rustup update` for updating immediately before running `cargo +nightly fmt` for formatting.

**Comment by athei** [comment]
What CI is telling you right now is that your companion for polkadot is failing to compile.

**Comment by ECJ222** [comment]
> What CI is telling you right now is that your companion for polkadot is failing to compile.

All the checks in my (Polkadot)companion pr passes, but the Polkadot CI check in this pr is still failing.

@athei please is it still for the same reason you mentioned earlier?

File: frame/treasury/src/lib.rs [hunk 8]
```diff
@@ -255,21 +255,20 @@ pub mod pallet {
...
-		/// Some funds have been allocated. \[proposal_index, award, beneficiary\]
-		Awarded(ProposalIndex, BalanceOf<T, I>, T::AccountId),
-		/// A proposal was rejected; funds were slashed. \[proposal_index, slashed\]
-		Rejected(ProposalIndex, BalanceOf<T, I>),
-		/// Some of our funds have been burnt. \[burn\]
-		Burnt(BalanceOf<T, I>),
+		/// New proposal.
+		Proposed { proposal_index: ProposalIndex },
+		/// We have ended a spend period and will now allocate funds.
+		Spending { funds: BalanceOf<T, I> },
```
Comment (kianenigma) [inline frame/treasury/src/lib.rs:261]: `budget_remaining` seems better to me.

File: frame/treasury/src/lib.rs [hunk 9]
```diff
@@ -255,21 +255,20 @@ pub mod pallet {
...
+		/// Some funds have been allocated.
+		Awarded { proposal_index: ProposalIndex, funds: BalanceOf<T, I>, account: T::AccountId },
```
Comment (kianenigma) [inline frame/treasury/src/lib.rs:263]:
```suggestion
		Awarded { proposal_index: ProposalIndex, award: BalanceOf<T, I>, account: T::AccountId },
```

**Review by kianenigma — COMMENTED** [review]
Mostly LGTM, jut 2 small notes. 

Polkadot companion is also broken it seems.

**Comment by athei** [comment]
> @athei please is it still for the same reason you mentioned earlier?

I seems that I misunderstood how the companion CI works these days. It appears that the failing job is expected when applying a breaking change. I will need to check whether it is OK to merge.

**Comment by ECJ222** [comment]
> I seems that I misunderstood how the companion CI works these days. It appears that the failing job is expected when applying a breaking change. I will need to check whether it is OK to merge.

Okay @athei.

**Review by athei — APPROVED** [review]

**Comment by KiChjang** [comment]
CI is not making any damn sense to me, I've restarted it multiple times but it still errors on the event types on the companion PR, but then the companion PR CI is all passing! I've no idea what's going on here, it's all arcane magic to me. cc @paritytech/ci

**Comment by bkchr** [comment]
bot merge

**Comment by bkchr** [comment]
bot merge force

**Comment by bkchr** [comment]
bot merge

**Comment by ECJ222** [comment]
Hi @bkchr, 

Hope you are having a great day, please do you know why this error occurs `Error: Response error (status 404 Not Found)` when you try the bot merge command?

**Comment by athei** [comment]
bot merge

**Pull request merged**